composer/trainer/trainer_hparams.py

```diff
@@ -85,7 +85,7 @@ def _initialize_dataloader(
             )
 
         train_device_batch_size = batch_size // dist.get_world_size()
-        if dataset_hparams.shuffle and subset_num_batches is not None:
+        if dataset_hparams.shuffle and subset_num_batches is not None and subset_num_batches != -1:
             warnings.warn(
                 (f'SubsetNumBatchesWarning: When specifying `subset_num_batches` for the {dataloader_label} dataset, '
```

Comment:
Should this be `[train|eval]_subset_num_batches` in the warning as for a user there is not variable named subset_num_batches?

Reply:
yes let me update